### PR TITLE
Carry #13: Add Docker rule to Makefile, update build documentation

### PR DESCRIPTION
**Pull request by samuelkarp**
*Issue #, if available:*
#13 

*Description of changes:*
Carry over PR #13 from @bgammill

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,6 +26,9 @@ firectl.sha256:
 firectl: $(SRCFILES)
 	go build
 
+docker:
+	docker run --rm -v ${PWD}:/firectl --workdir /firectl golang:1.11 make
```
Comment (nmeyerhans) [inline Makefile]: Any reason we shouldn't be using Go 1.12 at this point?

File: Makefile [hunk 2]
```diff
@@ -26,6 +26,9 @@ firectl.sha256:
 firectl: $(SRCFILES)
 	go build
 
+docker:
```
Comment (nmeyerhans) [inline Makefile]: Should this be called `build-in-docker` or something a little more descriptive? If I just saw the target name, I might think this was building a container image or something.

File: Makefile (hunk 1 above)
Comment (samuelkarp) [reply Makefile]: No particular reason except that our stated requirement is 1.11.  I can update.

File: Makefile (hunk 2 above)
Comment (samuelkarp) [reply Makefile]: Sure.

**Pull request merged**